CRM/Twingle/Exceptions/BaseException.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+use CRM_Twingle_ExtensionUtil as E;
+
+/**
+ * A simple custom exception class that indicates a problem within a class
+ * of the Twingle API extension.
+ */
+class CRM_Twingle_Exceptions_BaseException extends Exception {
+
+  private string $error_code;
```

Comment:
How will this be different from the base class's [`Exception::$code`](https://www.php.net/manual/en/class.exception.php#exception.props.code), i. e. why not just overriding this property and type it as `int|string`, as is done e. g. in [`PDOException`](https://www.php.net/manual/en/class.pdoexception.php#pdoexception.props.code)?

Reply:
Well, I think it's a matter of taste. I think, overriding doesn't provide any benefit here neither and this way the code is more explicit, because all the custom error codes are strings anyway. But I would be fine with overriding, too.